docs/4.x/archiving.md

```diff
@@ -137,10 +153,165 @@ For example, the *UserSettings* plugin uses one record to hold browser details o
+if the user requests them.**
+
+## Specifics of archive querying
+
+Archive data is queried through the `Archive` class:
+
+```
+$archive = Archive::build($idSite, $period, $date, $segment);
+$dataTable = $archive->getDataTableFromNumeric(['nb_visits', 'nb_visits_converted']);
+```
+
+When a query is made, the class looks for usable archives for the given `idSite`, `period` and `segment`. This will include:
+
+* the latest "all plugins" archive (where the done flag is like `done`) that we are allowed to query OR
+  the latest plugin specific archives (where the done flag is like `done.MyPlugin`) that would have the specific reports/metrics we are looking for
+* any partial archives for this `idSite`/`period`/`segment` combination that are newer than the latest archive above
+
+`Archive` will then look for the requested data in all of these archives and use the most recently archived data (ie, the rows with the greatest `ts_archived`).
+
+## Archive Purging
+
+Archives that are invalidated or no longer require processing do not need to be stored, so they are deleted. Old and
+errored archives are deleted in two places:
+
+* right after a new archive is finalized, older archives for the same site, period and segment are deleted
+* regularly through a scheduled task in the `CoreAdminHome` plugin, older and errored archives are deleted
+
+The specifics of purging are straightforward. `idarchive`s of archives that are safe to delete are queried, then
+rows with that `idarchive` in the respective numeric and blob tables are deleted.
+
+### Other instances where archives are deleted
+
+Archives can also be deleted in other contexts. For example, when a segment is deleted, we no longer need the
+archives for that segment, so we delete them.
+
+Plugins can delete archives themselves by using the `ArchivePurger` class.
+
+## Partial Archives
+
+Partial archives are a special type of archive that only contain one or a few reports of one plugin, rather than every report
+for one plugin (or every report for all plugins). These archives have done flag value of `ArchiveWriter::DONE_PARTIAL`.
+
+Currently, they are only created when plugins want to archive a single report in the past.
+
+### For plugins: supporting partial archives
+
+Partial archives are not automatically created when a single report is requested for archiving. Plugin `Archiver`s have to
+specifically support this workflow like so:
+
+```
+class Archiver extends \Piwik\Plugin\Archiver
+{
+    // ...
+
+    public function __construct(ArchiveProcessor $processor)
+    {
+        parent::__construct($processor);
+
+        // if a single report is requested, mark the archive we're creating as partial
+        $this->requestedReport = $processor->getParams()->getArchiveOnlyReport();
+        if ($this->requestedReport) {
+            $processor->getParams()->setIsPartialArchive(true);
```

Comment:
@diosmosis didn't notice this was needed? Any chance we could handle this automatic? (`$processor->getParams()->setIsPartialArchive(true);`)

Reply:
No, because the extra code in aggregateDay/MultiplePeriods has to exist too, otherwise we'd mark a full archive as partial. It has to be done by Archivers that specifically support this behavior.

Reply:
OK. I thought it would in worst case only end up with only a done flag in the archive but no actual report (or all reports archived again instead of the specifically requested report in that plugin).